### PR TITLE
Separates game-logic from user-interface

### PR DESCRIPTION
**Pull request by fc-anjos**
This pull request merges game-logic branch, effectively separating the user interaction from the logic itself. attr-accessors and attr-readers are used where needed.

### CODE REVIEW

**Comment by Anna-Myzukina** [comment]
# Status: [changes required]


### Good job with the milestone. Few steps and your project can be approved:
`Implementation details`


*  The code of the project should be logically divided between classes.(Usually it uses three: Game, Board and Player) from [Pathwright](https://microverse.pathwright.com/library/fast-track-curriculum/69047/path/step/59565018/). 
screenshot 

* All classes are kept in lib directory from [Pathwright](https://microverse.pathwright.com/library/fast-track-curriculum/69047/path/step/59565018/). 
screenshot 

* The code should respect the single responsibility rule for methods (i.e. each method does exactly one thing). You don`t need to create long methods that are doing multiple steps of the game process from [Pathwright](https://microverse.pathwright.com/library/fast-track-curriculum/69047/path/step/59565018/). 
screenshot 

### Make changes and submit another code review request.

*If you have any questions, let me know on slack (@Anna-Muzykina) and I will be glad to clarify with you.*
*Happy coding * <g-emoji class="g-emoji" alias="computer" fallback-src="https://github.githubassets.com/images/icons/emoji/unicode/1f4bb.png">computer</g-emoji>

**Review by mosesogwo — CHANGES_REQUESTED** [review]
**Hi @fc-anjos, @thankgodr :clap:. Some changes are still required :fast_forward:**

- Move the String class from main.rb into a separate file in lib folder.
- A class isn't required for the game interface in main.rb. You should reorganize your code in main.rb to use a clear loop that calls the game actions for one round and repeat them for each player. This will help you simplify some of the methods in main.rb.

**You have just three commits in this milestone. You should commit more regularly.**

_**Kindly make these changes and submit another code review request.**_

**Any question? You can reach me on Slack (Moses Ogwo)**

**Cheers:stars:**

### Status: Changes Requested.

**Comment by pelzolga123** [comment]
## Hi Team!  
### Your project looks :+1: amazing!
Just one more thing looks not right:

* [ ] After there is draw, game doesn't display message about it, but immediately starts new game. That needs fixing.

### Please kindly fix the above issue and send another code review request.

If you have any questions or need help, you can message me on slack (@ Olga Pelts). *Happy coding*

File: lib/board.rb [hunk 1]
```diff
@@ -0,0 +1,17 @@
...
+    update_board(@arr)
+  end
+
+  def update_board(arr)
+    @arr = arr
+    @board = "     #{@arr[0][0]} | #{@arr[0][1]} | #{@arr[0][2]}
+     ___*___*___
+     #{@arr[1][0]} | #{@arr[1][1]} | #{@arr[1][2]}
+     ___*___*___
+     #{@arr[2][0]} | #{@arr[2][1]} | #{@arr[2][2]} "
```
Comment (euqueme) [inline lib/board.rb:15]: [OPTIONAL]

- [ ] I think the code of the @board variable in your update_board method is a little repetitive and could be achieved with a loop, so I strongly recommend to change it to keep your code DRY.

**Review by euqueme — APPROVED** [review]
> ### Pull Request under the review by Maria Eugenia Queme P. (TSE)

### Hello @fc-anjos and @thankgodr :balloon: :lollipop: 

### You've done a great job with your game, thank you for doing all the changes requested by the previous reviewers :medal_sports:  I think your game is good enough to pass :heavy_check_mark: 

## OPTIONAL

#### Behaviour

- [ ] The game allows you to have empty names for the players, so I strongly recommend to either add a validation when no player name is given or set a default name when this happens
image

#### Implementation

* [ ] The code should respect the single responsibility rule for methods (i.e. each method does exactly one thing). ([screenshot](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/ruby/OOP/milestone3/single_responsibility.png) from [Pathwright](https://microverse.pathwright.com/library/fast-track-curriculum/69047/path/step/59565018/))

- You have methods in your main.rb file that have multiple responsibilities, they have loops and are requesting for input, validating and printing values, I strongly recommend to give all methods a single responsibility and just keep the game loops outside the methods.

### That's it, team! :raised_hands: 

### Please :pray: do the optional changes (or not) :man_technologist: :man_technologist:  and merge this branch when you are done :memo: 

Happy coding! :rocket: 

Any questions? ask me on slack **@ Maru** :woman_technologist: 

PROJECT STATUS: APPROVED :green_apple:

**Pull request merged**